PixelHunter1995/Player.cs

```diff
@@ -64,18 +87,21 @@ public void HandleInput(InputManager input)
         {
             if (input.GetState(InputCommand.PLAYING_Move).IsDown)
             {
-                // Compensate for Position being in top left corner
-                float x = input.MouseSceneX - AnimationTileset.tileWidth / 2;
-                float y = input.MouseSceneY - AnimationTileset.tileHeight;
-                this.MovePosition = new Vector2(x, y);
-            }
-            if (input.Input.GetKeyState(MouseKeys.LeftButton).IsEdgeDown)
-            {
-                Say("Hi, I'm the player!");  // For debugging purposes, have the character talk when walking
-                Say("Well, actually my name is Felixia.");
-                Say("My interests include alchemy, solving ridiculously convoluted puzzles " +
-                    "and long pull requests on the shore.");
+                float x = input.MouseSceneX;
+                float y = input.MouseSceneY;
+                // Check if position is in Scene (and not inventory etc)
+                if (0 < y && y < GlobalSettings.SCENE_HEIGHT)
+                {
+                    LastClickedPosition = new Vector2(x, y);
+                    Say("Hi, I'm the player!");  // For debugging purposes, have the character talk when walking
```

Comment:
With this you will queue up a new set of things to say every frame that the left click is down. That was why it was inside an IsEdgeDown call. But maybe you can just remove this code.

Reply:
And also I think we should listen for IsEdgeDown when walking in any case, since that is how it works in Monkey Island :P